packages/google_maps_flutter/ios/Classes/GoogleMapMarkerController.m

```diff
@@ -61,3 +70,172 @@ - (void)setZIndex:(int)zIndex {
+                                   NSObject<FlutterPluginRegistrar>* registrar) {
+  NSDictionary* data = json;
+  id alpha = data[@"alpha"];
+  if (alpha) {
+    [sink setAlpha:toFloat(alpha)];
+  }
+  id anchor = data[@"anchor"];
+  if (anchor) {
+    [sink setAnchor:toPoint(anchor)];
+  }
+  id draggable = data[@"draggable"];
+  if (draggable) {
+    [sink setDraggable:toBool(draggable)];
+  }
+  id icon = data[@"icon"];
+  if (icon) {
+    NSArray* iconData = icon;
+    UIImage* image;
+    if ([iconData[0] isEqualToString:@"defaultMarker"]) {
+      CGFloat hue = (iconData.count == 1) ? 0.0f : toDouble(iconData[1]);
+      image = [GMSMarker markerImageWithColor:[UIColor colorWithHue:hue / 360.0
+                                                         saturation:1.0
+                                                         brightness:0.7
+                                                              alpha:1.0]];
+    } else if ([iconData[0] isEqualToString:@"fromAsset"]) {
+      if (iconData.count == 2) {
+        image = [UIImage imageNamed:[registrar lookupKeyForAsset:iconData[1]]];
+      } else {
+        image = [UIImage imageNamed:[registrar lookupKeyForAsset:iconData[1]
+                                                     fromPackage:iconData[2]]];
+      }
+    }
+    [sink setIcon:image];
+  }
+  id flat = data[@"flat"];
+  if (flat) {
+    [sink setFlat:toBool(flat)];
+  }
+  id consumeTapEvents = data[@"consumeTapEvents"];
+  if (consumeTapEvents) {
+    [sink setConsumeTapEvents:toBool(consumeTapEvents)];
+  }
+  id infoWindow = data[@"infoWindow"];
+  if (infoWindow) {
+    NSString* title = infoWindow[@"title"];
+    NSString* snippet = infoWindow[@"snippet"];
+    if (title) {
+      [sink setInfoWindowTitle:title snippet:snippet];
+    }
+    id infoWindowAnchor = infoWindow[@"infoWindowAnchor"];
+    if (infoWindowAnchor) {
+      [sink setInfoWindowAnchor:toPoint(infoWindowAnchor)];
+    }
+  }
+  id position = data[@"position"];
+  if (position) {
+    [sink setPosition:toLocation(position)];
+  }
+  id rotation = data[@"rotation"];
+  if (rotation) {
+    [sink setRotation:toDouble(rotation)];
+  }
+  id visible = data[@"visible"];
+  if (visible) {
+    [sink setVisible:toBool(visible)];
+  }
+  id zIndex = data[@"zIndex"];
+  if (zIndex) {
+    [sink setZIndex:toInt(zIndex)];
+  }
+}
+
+@implementation FLTMarkersController {
+  NSMutableDictionary* _markerIdToController;
+  FlutterMethodChannel* _methodChannel;
+  NSObject<FlutterPluginRegistrar>* _registrar;
+  GMSMapView* _mapView;
+}
+- (instancetype)init:(FlutterMethodChannel*)methodChannel mapView:(GMSMapView*)mapView {
+  self = [super init];
+  if (self) {
+    _methodChannel = methodChannel;
+    _mapView = mapView;
+    _markerIdToController = [NSMutableDictionary dictionaryWithCapacity:1];
+  }
+  return self;
+}
+- (void)addMarkers:(id)markersToAdd {
+  NSArray<id>* markers = markersToAdd;
+  if (!markers) {
```

Comment:
A safer check would be:
`if (![markersToAdd isKindOfClass:[NSArray class]])`, this also covers the cases that it's NSNull (if it was explicitly set to null), or that it is a wrong type.